### PR TITLE
Remove use of deepcopy

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  

## What does this PR do?
After profiling some metrics, I saw that that `reset` is spending a lot of time calling `deepcopy`. As the recommended method for cloning a tensor is `.detach().clone()` and it is a bit faster:
```
master:
tensor state (cpu): 0.8742028713226319+-0.03886386210968593
tensor state (gpu): 2.5115918636322023+-0.21529070194941985
list state (cpu): 2.109953451156616+-0.07316269549459964
list state (gpu): 1.809204387664795+-1.0394736704878353

PR:
tensor state (cpu):0.8499699592590332+-0.01647528474876866
tensor state (gpu):2.2886361598968508+-0.08935389958063863
list state (cpu): 2.0209419250488283+-0.05932843043190113
list state (gpu): 1.547454833984375+-0.407560693521206
```
calculate for 5 repetitions over 1000 forward calculations (`reset` is called once for every `forward` call which is why such a small optimization may matter in the long run). 

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: torchmetrics/metric.py [hunk 1]
```diff
@@ -259,9 +259,9 @@ def reset(self):
         for attr, default in self._defaults.items():
             current_val = getattr(self, attr)
             if isinstance(default, Tensor):
-                setattr(self, attr, deepcopy(default).to(current_val.device))
+                setattr(self, attr, default.detach().clone().to(current_val.device))
             else:
-                setattr(self, attr, deepcopy(default))
+                setattr(self, attr, [])
```
Comment (Borda) [inline torchmetrics/metric.py:264]: we do not need default anymore?

Comment (SkafteNicki) [reply torchmetrics/metric.py:264]: since the default value of metric states can either be tensors or an empty list, in this case it always be an empty list

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by justusschock — APPROVED** [review]
